src/librustdoc/clean/mod.rs

```diff
@@ -1143,7 +1143,7 @@ fn clean_fn_decl_with_args<'tcx>(
     decl: &hir::FnDecl<'tcx>,
     header: Option<&hir::FnHeader>,
     args: Arguments,
-) -> FnDecl {
+) -> FunctionSignature {
```

Comment:
> The reason for renaming FnDecl, and the whole PR in general, is to be more friendly to outside users, that might not be in the know about rustc conventions, a function declaration may be understood as fn printf(fmt: &str, ...);,even though what the FnDecl is there is actually just (fmt: &str, ...), which is universally understood to be a function signature

am I right in thinking that anyone using this output will know enough rust to relate the `fn` keyword to functions, and probably the `Fn*` traits.

Reply:
well, `fn` is a fundamental keyword in Rust, so `fn` <-> `function` is a trivial expectation from the user, the change from `Fn` to `Function` there is purely for consistency with the existing `Function` & `FunctionPointer` structures, I don't believe we should be swayed by rustc's naming in rustdoc-json, since rustc is definitely not intended to be public API, unlike rustdoc-json

Reply:
And the change from `Decl` to `Signature` is explained in my comment a bit below